### PR TITLE
checking rancher settings are not empty

### PR DESCRIPTION
**Pull request by brudnak**
## Problem
In Rancher versions 2.8.3 the webhook version, fleet version, and CSP adapter version are not set in the settings or environment variables. This allows these components to be automatically upgraded to newer versions, which may cause unexpected behavior or compatibility issues.

## Solution
From a QA perspective, the solution involves adding automated tests to ensure that the versions for the webhook, fleet, and CSP adapter not empty. The tests will:
- Retrieve the settings for `rancher-webhook-version`, `fleet-version`, and `csp-adapter-min-version` using `kubectl` commands.
- Verify that the `source` field in the settings is set to `"env"`.
- Check that the `value` field in the settings is not empty.

These tests will be implemented using the Go testing framework and the Rancher Shepherd library to interact with Rancher and perform the necessary assertions.

### Automated Testing
* Test types added/modified:
   * Validation (Go Framework)
Summary:
- Updated a test suite `WebhookTestSuite` with a new test case `TestWebhookPinnedVersion` to verify that the webhook version is pinned and not empty in the settings and environment variable.
- Added a test suite `CSPTestSuite` with a test case `TestCSPAdapterMinVersionNotEmpty` to ensure that the CSP adapter minimum version is set and not empty in the settings and environment variable.
- Added a test suite `FleetTestSuite` with a test case `TestFleetVersionNotEmpty` to check that the fleet version is specified and not empty in the settings and environment variable.

### CODE REVIEW

File: tests/v2/validation/envvars/build_versions_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	"github.com/stretchr/testify/suite"
+	"testing"
+
+	"github.com/rancher/shepherd/clients/rancher/catalog"
+	"github.com/rancher/shepherd/extensions/charts"
+
+	"github.com/rancher/shepherd/extensions/clusters"
+)
+
+type BuildVersionsTestSuite struct {
```
Comment (Priyashetty17) [inline tests/v2/validation/envvars/build_versions_test.go:21]: Previously, there was a discussion about reducing the types defined in the global struct. Can we limit it to just these and move the rest within the individual tests?
```
client *rancher.Client
session *session.Session
cluster *management.Cluster
```

Also, looks like steveClient isn't used anywhere.

File: tests/v2/validation/envvars/build_versions_test.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	fleetVersion         string
+}
+
+func (b *BuildVersionsTestSuite) TearDownSuite() {
+	b.session.Cleanup()
+}
+
+func (b *BuildVersionsTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	b.session = testSession
```
Comment (Priyashetty17) [inline tests/v2/validation/envvars/build_versions_test.go:40]: Can you change this to: 
`b.session = session.NewSession()`

File: tests/v2/validation/envvars/build_versions_test.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+	buildYamlContent, err := getBuildYamlContent(majorMinorVersion)
+	require.NoError(b.T(), err)
+
+	b.webhookVersion, b.cspAdapterMinVersion, b.fleetVersion, err = extractBuildValues(buildYamlContent)
+	require.NoError(b.T(), err)
+}
+
+func (b *BuildVersionsTestSuite) TestWebhookVersion() {
+	settingName := "rancher-webhook-version"
```
Comment (Priyashetty17) [inline tests/v2/validation/envvars/build_versions_test.go:68]: Can this be a constant instead?

File: tests/v2/validation/envvars/build_versions_test.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	actualVersion, err := b.client.Management.Setting.ByID(settingName)
+	require.NoError(b.T(), err)
+	require.Equal(b.T(), b.webhookVersion, actualVersion.Value)
+}
+
+func (b *BuildVersionsTestSuite) TestCSPAdapterMinVersion() {
+	settingName := "csp-adapter-min-version"
```
Comment (Priyashetty17) [inline tests/v2/validation/envvars/build_versions_test.go:75]: Can this be a constant instead?

File: tests/v2/validation/envvars/build_versions_test.go [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+	actualVersion, err := b.client.Management.Setting.ByID(settingName)
+	require.NoError(b.T(), err)
+	require.Equal(b.T(), b.cspAdapterMinVersion, actualVersion.Value)
+}
+
+func (b *BuildVersionsTestSuite) TestFleetVersion() {
+	settingName := "fleet-version"
```
Comment (Priyashetty17) [inline tests/v2/validation/envvars/build_versions_test.go:82]: Can this be a constant instead?

File: tests/v2/validation/envvars/build_versions_test.go [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+	b.session.Cleanup()
+}
+
+func (b *BuildVersionsTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	b.session = testSession
+	client, err := rancher.NewClient("", b.session)
+	require.NoError(b.T(), err)
+	b.client = client
+	b.steveClient = client.Steve
```
Comment (Priyashetty17) [inline tests/v2/validation/envvars/build_versions_test.go:44]: Can you remove this since this isn't used anywhere?

**Review by Priyashetty17 — COMMENTED** [review]
Have a few comments

File: tests/v2/validation/envvars/build_versions_test.go (hunk 1 above)
Comment (brudnak) [reply tests/v2/validation/envvars/build_versions_test.go]: removed, thank you!

File: tests/v2/validation/envvars/build_versions_test.go (hunk 2 above)
Comment (brudnak) [reply tests/v2/validation/envvars/build_versions_test.go]: changed, thanks! 😁

File: tests/v2/validation/envvars/build_versions_test.go (hunk 3 above)
Comment (brudnak) [reply tests/v2/validation/envvars/build_versions_test.go]: moved to a constant, ty!

File: tests/v2/validation/envvars/build_versions_test.go (hunk 4 above)
Comment (brudnak) [reply tests/v2/validation/envvars/build_versions_test.go]: moved to a constant, ty!

File: tests/v2/validation/envvars/build_versions_test.go (hunk 5 above)
Comment (brudnak) [reply tests/v2/validation/envvars/build_versions_test.go]: moved to a constant, ty!

File: tests/v2/validation/envvars/build_versions_test.go (hunk 6 above)
Comment (brudnak) [reply tests/v2/validation/envvars/build_versions_test.go]: removed, ty!

**Review by dasarinaidu — APPROVED** [review]
LGTM:)
